### DISCUSSION TITLE
Support custom engine and mod dlls

### DISCUSSION THREAD

**Discussion by UnkwUsr** [JamesIves/hlds-docker#29] [Ideas and Feature Requests] [answered]
What about supporting custom engine (like [rehlds](https://github.com/dreamstalker/ReHLDS) (only one I know)) and custom mod dlls (like for valve - [BugfixedHL](https://github.com/tmp64/BugfixedHL-Rebased))?
Regarding first, I think it is not possible yet (UPD: probably with building custom image?). Regarding second, will it work if I put `dlls/hl.so` in `config/`?

**Comment by JamesIves** [comment]
I don't see any reason why these wouldn't work already using some combination of the advanced setup options mentioned in the getting started guide. Whatever you place in the config directory will map directly onto the server client within the mod directory, so placing something in config\dlls should work as it will end up in `{GAME_DIR}/dlls/*`. I've done something similar when running a modded version of Half-Life Decay where I've added custom dlls, and added the associated game files to point to them.

You might need to build a custom image for the first item though if you need to download a very specific build of Half Life for it to work however, but after following that Readme the requirements are not super clear. Is the legacy engine version delivered as a beta client named steam_legacy? If this is a common use case I'd be open to providing pre built images for this (ie valve-steam-legacy) etc. 

**Edit: I did some more research on this, found the patch notes from Valve. Here's some new features to help with this that have just been merged:**

* When building a custom image, you can now use an environment variable called `FLAG`, ie `export FLAG=-beta steam_legacy`
* Configured pre-built legacy images for Half-Life Deathmatch, Counter-Strike and Condition Zero, these are available via `jives/hlds:valve-legacy`, `jives/hlds:cstrike-legacy` and `jives/hlds:czero-legacy`. From what I can tell only these three titles have legacy versions at this time.

**Reply by JamesIves** [reply, marked as answer]
Release notes: https://github.com/JamesIves/hlds-docker/releases/tag/v1.0.18
